pkg/vetter/conflictingvirtualservicehost/vet.go

```diff
@@ -392,6 +398,22 @@ func (v *VsHost) Vet() ([]*apiv1.Note, error) {
 	return notes, nil
 }
 
+// check same VS route conflict
+func sameVSconflict(rule1 routeRule, rule2 routeRule) (bool, error) {
+
+	if rule1.vsName == rule2.vsName && rule1.namespace == rule2.namespace {
+		if (rule1.priority > rule2.priority &&
+			rule1.ruleType <= rule2.ruleType) {
```

Comment:
I'd advocate for splitting this out into specific cases of the different rule types instead of using that it's an enum. While it works, it's not at all easy to reason about and you have to navigate to the enum definition to see in what order the rule types were defined.

Reply:
hmm, hate to hard code this and later add another ruletype. Maybe best to put comment in the enum definition.

Reply:
I don't think commenting the enum definition would solve the problem of needing to check the order of the enum declaration when looking at this line.

I don't see this having any benefit to having the hard-coded replies. This is potentially more dangerous if we add a new type, since it can silently do the wrong thing unless we explicitly check for this, anyway.

See: https://github.com/aspenmesh/istio-vet/blob/master/pkg/vetter/conflictingvirtualservicehost/vet.go#L365 as an example that would handle your case of adding a route rule. It explicitly reports an error so that it is obvious where the error is coming from and it's easy to fix. With this method, we could get a wrong answer and have no idea why.

Of all of my comments, this is the one I feel most strongly about.